oauth/resource_service_principal_federation_policy.go

```diff
@@ -0,0 +1,140 @@
+package oauth
+
+import (
+	"context"
+	"github.com/databricks/databricks-sdk-go/service/oauth2"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"log"
+	"strings"
+)
+
+func ResourceServicePrincipalFederationPolicy() common.Resource {
+	s := map[string]*schema.Schema{
+		"description": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"service_principal_id": {
+			Type:     schema.TypeInt,
+			Required: true,
+			ForceNew: true,
+		},
+		"name": {
+			Type:     schema.TypeString,
+			Optional: true,
+			ForceNew: true,
+			//Computed: true,
+		},
+		"uid": {
+			Type:     schema.TypeString,
+			Computed: true,
+			Optional: true,
+		},
+		"oidc_policy": {
+			Type:     schema.TypeSet,
+			Required: true,
+			MinItems: 1,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"audiences": {
+						Type:     schema.TypeList,
+						Optional: true,
+						Elem: &schema.Schema{
+							Type: schema.TypeString,
+						},
+					},
+					"issuer": {
+						Type:     schema.TypeString,
+						Required: true,
+					},
+					"jwks_json": {
+						Type:     schema.TypeString,
+						Optional: true,
+					},
+					"jwks_uri": {
+						Type:     schema.TypeString,
+						Optional: true,
+					},
+					"subject": {
+						Type:     schema.TypeString,
+						Required: true,
+					},
+					"subject_claim": {
+						Type:     schema.TypeString,
+						Optional: true,
+					},
+				},
+			},
+		},
+	}
+	return common.Resource{
+		Schema: s,
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var federationPolicy oauth2.FederationPolicy
+			common.DataToStructPointer(d, s, &federationPolicy)
+			acc, err := c.AccountClient()
+			if err != nil {
+				return err
+			}
+			spfp, err := acc.ServicePrincipalFederationPolicy.Create(ctx,
+				oauth2.CreateServicePrincipalFederationPolicyRequest{
+					Policy:             &federationPolicy,
+					ServicePrincipalId: int64(d.Get("service_principal_id").(int)),
+				})
+			if err != nil {
+				return err
+			}
+			name := spfp.Name
+			d.SetId(name[strings.LastIndex(name, "/")+1:])
```

Comment:
I'm not sure I understand this. If name is optional and can be user-specified then I don't think we can assume it is the ID ("uid" field). Though, Databricks docs say "if specified in a request, must match the value in the request URL", but they must be referring to account ID and SP ID because policy ID is not in the URL and is not known until the policy is created.

https://docs.databricks.com/api/account/serviceprincipalfederationpolicy/create

It seems simpler to just use the uid field, directly, since this is what the API expects for update/delete requests, not necessarily the last part of the name.

https://pkg.go.dev/github.com/databricks/databricks-sdk-go@v0.64.0/service/oauth2#FederationPolicy

```suggestion
			d.SetId(spfp.Uid)
```

Reply:
Oh, you are right, I just double checked with the CLI. Thanks for explaining! It's quite strange because Databricks calls this part "policy id" in their API docs, but the examples suggest it is the end part of the name, like you say.

https://docs.databricks.com/api/account/serviceprincipalfederationpolicy/get

Reply:
I’ve updated the code so that the UUID is used as the ID and is unique. I’d really appreciate it if you could review it